remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/mapping/model/ProtobufRawTypeModel.java

```diff
@@ -64,6 +66,11 @@ public Collection<PojoPropertyModel<?>> declaredProperties() {
       return Collections.emptySet();
    }
 
+   @Override
+   public PojoTypeModel<? extends byte[]> cast(PojoTypeModel<?> other) {
+      return null;
```

Comment:
`return this` would be more in line with the javadoc of this method.
And if you want to be really correct, the right implementation would be that of `HibernateOrmDynamicMapRawTypeModel`:

```java
		if ( other.rawType().isSubTypeOf( this ) ) {
			// Redundant cast; no need to create a new type.
			return (PojoTypeModel<? extends Map>) other;
		}
		else {
			// There is no generic type information to retain for protobuf types; we can just return this.
			// Also, calling other.castTo(...) would mean losing the type name, and we definitely don't want that.
			return this;
		}
```

Reply:
Thanks @yrodiere, my implementation would be:
``` java
if ( other.rawType().isSubTypeOf( this ) ) {
 // Redundant cast; no need to create a new type.
 return (PojoTypeModel<? extends byte[]>) other;
}
else {
 // There is no generic type information to retain for protobuf types; we can just return this.
 // Also, calling other.castTo(...) would mean losing the type name, and we definitely don't want that.
 return this;
}
```

Because we need to return a `PojoTypeModel<? extends byte[]>`. Is that correct? Thanks

Reply:
Right, I forgot to replace `Map`. Yes, that's correct.